### PR TITLE
Forbid expensive query parts in ranking evaluation

### LINKED ISSUE #29674: Remove unneeded elements from ranking evaluation queries
Currently the ranking evaluation API accepts the full query syntax for the queries specified in the test set and executes them as a multi search. This potentially runs costly aggregations and suggestions too. Since I think aggregations, suggestions and highlighting don't play a role in search evaluation, we ignore and remove those parts of the query if they get accidentally added in order to reduce its cost.
I'm not sure yet if it would be better to throw an error in these cases or just silently ignore the parts
that are irrelevent in the context of this API.

### PR DESCRIPTION
**Pull request by cbuescher**
Currently the ranking evaluation API accepts the full query syntax for
the queries specified in the evaluation set and executes them via multi
search. This potentially runs costly aggregations and suggestions too.
This change adds checks that forbid using aggregations, suggesters or
highlighters in the queries that are run as part of the ranking
evaluation since they are irrelevent in the context of this API.

Closes #29674

### CODE REVIEW

**Comment by cbuescher** [comment]
@elasticmachine run sample packaging tests

File: modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RatedRequest.java [hunk 1]
```diff
@@ -129,18 +154,24 @@ private RatedRequest(String id, List<RatedDocument> ratedDocs, SearchSourceBuild
...
-    public RatedRequest(String id, List<RatedDocument> ratedDocs, Map<String, Object> params,
-            String templateId) {
-        this(id, ratedDocs, null, params, templateId);
-    }
-
-    public RatedRequest(String id, List<RatedDocument> ratedDocs, SearchSourceBuilder testRequest) {
-        this(id, ratedDocs, testRequest, new HashMap<>(), null);
+    static void validateEvaluatedQuery(SearchSourceBuilder evaluationRequest) {
+        // ensure that testRequest, if set, does not contain aggregation, suggest or highlighting section
+        if (evaluationRequest != null) {
```
Comment (polyfractal) [inline modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RatedRequest.java]: I wonder if `profile` and `explain` should be forbidden too?  Both have non-negligible impact on performance, and seem irrelevant to ranking as well.

**Comment by polyfractal** [comment]
Left a minor comment but otherwise LGTM :)

File: modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RatedRequest.java (hunk 1 above)
Comment (cbuescher) [reply modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RatedRequest.java]: Good point, I think I will add that before I merge. Thanks.
